### PR TITLE
Add btc_price api for getting the cached exchange.

### PR DESCRIPTION
**Pull request by ddustin**
Also return a list of known currencyies with exchange rates.

Example usage: /api/v1/btc_price?currency=USD

### CODE REVIEW

File: api/restapi.py [hunk 1]
```diff
@@ -1327,6 +1328,17 @@ def get_node(node):
             request.finish()
         return server.NOT_DONE_YET
 
+    @GET('^/api/v1/btc_price')
```
Comment (cpacia) [inline api/restapi.py]: just want to add the @authenticated decorated here.

**Comment by saltduck** [comment]
Why not return multiple exchange rates while given multiple currency parameters? Just like following:

    /api/v1/btc_price?currency=USD&currency=CNY
```json
{"BtcExchange": {"USD": 414.0, "CNY": 2713}}
```

**Comment by cpacia** [comment]
The purpose of this api is to just send the rate used by the server to the client so they are both in sync. The client doesn't need to know multiple rates.

**Comment by cpacia** [comment]
Misread that comment. I suppose the api could return multiple rates, but the client only uses the currency you set in the settings. Unless someone else has need for multiple rates.

**Comment by jjeffryes** [comment]
The client will call this API for each item that's in a fiat currency, which could be a lot of calls on the discover page. It has to convert from the original fiat currency to btc and then to the user's fiat (if they aren't using btc). 

It would probably simplify things to just return an array of all currencies and their exchange rates, rather than one currency and then an array of currency codes. Then the client could store that array and not need to call the api again.

**Comment by cpacia** [comment]
@jjeffryes this PR will return all currencies if you just don't send the currency code.

**Pull request merged**